server/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -79,9 +94,12 @@
      * @param maxSeqNo         the current maximum sequence number of all operations in the translog
      * @param globalCheckpoint the last-known global checkpoint
      * @param minTranslogGeneration the minimum generation referenced by the translog at this moment.
+     * @param trimmedAboveSeqNo   the current maximum reachable (trimmed) sequence number of all operations in the translog
```

Comment:
How about: `all operations with seq# above trimmedAboveSeqNo should be ignored and not read from the corresponding translog file` . Also, there's some layouting issues.